owner/src/main/java/org/the_chance/honeymart/ui/features/signup/SignUpViewModel.kt

```diff
@@ -106,43 +40,122 @@ class SignUpViewModel @Inject constructor(
-            login(email = _state.value.email, password = _state.value.password)
-        }
         _state.update { it.copy(isLoading = false, isSignUp = isSignUp) }
     }
 
     private fun onError(error: ErrorHandler) {
-        _state.update {
-            it.copy(
-                isLoading = false, error = error,
-                fullNameState = ValidationState.INVALID_FULL_NAME,
-                emailState = ValidationState.INVALID_EMAIL
+        _state.update { it.copy(isLoading = false, error = error) }
+    }
+
+    override fun onClickLogin() {
+        effectActionExecutor(_effect, SignupUiEffect.ClickLoginEffect)
+    }
+
+    override fun onClickContinue() {
+        val validationState = state.value.emailState.isValid &&
+                state.value.passwordState.isValid &&
+                state.value.fullNameState.isValid &&
+                state.value.confirmPasswordState.isValid
+
+        if (validationState) {
+            addOwner(
+                fullName = state.value.fullNameState.value,
+                email = state.value.emailState.value,
+                password = state.value.passwordState.value
             )
+            if (state.value.isSignUp) {
+                effectActionExecutor(_effect, SignupUiEffect.ClickContinueEffect)
+            }
+        } else {
+            effectActionExecutor(_effect, SignupUiEffect.ShowValidationToast)
         }
     }
 
-
-    private fun login(email: String, password: String) {
-        _state.update { it.copy(isLoading = true) }
-        tryToExecute(
-            { loginOwner(password = password, email = email) },
-            ::onLoginSuccess,
-            ::onLoginError,
+    override fun onFullNameInputChange(fullName: CharSequence) {
+        val fullNameState = validationSignupFields.validationFullName(
+            fullName.trim().toString()
         )
+        _state.update {
+            it.copy(
+                fullNameState = FieldState(
+                    errorState = when (fullNameState) {
+                        ValidationState.BLANK_FULL_NAME -> "name shouldn't be empty"
+                        ValidationState.INVALID_FULL_NAME -> "Invalid name"
+                        else -> ""
+                    },
+                    value = fullName.toString(),
+                    isValid = fullNameState == ValidationState.VALID_FULL_NAME
+                ),
+            )
+        }
     }
 
-    private fun onLoginSuccess(loginState: ValidationState) {
-        if (loginState == ValidationState.SUCCESS)
-            effectActionExecutor(_effect, SignupUiEffect.ClickSignupEffect)
+    override fun onEmailInputChange(email: CharSequence) {
+        val emailState = validationSignupFields.validateEmail(email.trim().toString())
+        _state.update {
+            it.copy(
+                emailState = FieldState(
+                    errorState = when (emailState) {
+                        ValidationState.BLANK_EMAIL -> "email shouldn't be empty"
```

Comment:
Why message validation hardcode can't translate can wrap with code

Reply:
This process is done within the view model which is not applicable to the use of R.string because it needs context as you know